winlogbeat/checkpoint/checkpoint.go

```diff
@@ -89,6 +89,12 @@ func NewCheckpoint(file string, maxUpdates int, interval time.Duration) (*Checkp
 		}
 	}
 
+	// Write the state file to verify we have have permissions.
+	err = c.flush()
```

Comment:
I just wonder if this is really platform independent way to detect we can not write the file. When opening with write permissions, it should already file on open if we haven't had persmissions, right?